cudax/include/cuda/experimental/__stf/internal/constants.cuh

```diff
@@ -77,10 +79,19 @@ inline const char* access_mode_string(access_mode mode)
       return "write";
     case access_mode::relaxed:
       return "relaxed"; // op ?
+    case access_mode::reduce_no_init:
+      return "reduce (no init)"; // op ?
+    case access_mode::reduce_do_init:
+      return "reduce (do init)"; // op ?
     default:
       assert(false);
       abort();
   }
 }
 
+class no_init
+{};
+class do_init
+{};
```

Comment:
Hopefully we can get rid of `no_init` because it's implicit and only have the user specify `do_init` when they want to do initialization.

Reply:
True ! I'll try to remove that thanks for the suggestion

Reply:
Actually i do need a type (it's a pair of op type, and a type which is either no_init or do_init), so the best is perhaps to expose no_init{} tag for user, then consider true_type and false_type. This avoids an explosion of our types.

Reply:
that's done